parquet-column/src/main/java/org/apache/parquet/filter2/predicate/Statistics.java

```diff
@@ -26,17 +28,37 @@
 public class Statistics<T> {
   private final T min;
   private final T max;
+  private final Comparator<T> comparator;
 
-  public Statistics(T min, T max) {
+  public Statistics(T min, T max, Comparator<T> comparator) {
```

Comment:
Currently, we don't have dependency on Yetus but have InterfaceAudience from 3 different dependencies (hadoop, hive, pig). It is not clear to me which pattern should we follow (having our own annotation implementation, use Yetus, use Hadoop etc.) So, I would leave the annotation to a separate change. (It might worth a JIRA for that.)
I'll put back the original constructor as deprecated for now a comment that the constructors are not for public use.

Reply:
-1 for introducing audience annotations. We don't use those in Parquet. We've been needing a push to clean up the public API for a while now, but introducing the annotations for single classes isn't going to help. If we are going to clean up the API, lets do that.

Reply:
We would like to introduce it for the whole API, but it's a big task to do in a single chunk, so I think a progressive approach would be a good start. It is much easier to decide the intended audience of a method when it gets added and both developer and reviewers have a context than later on when it requires a massive investigation. For this reason I think it would be good to start adding these annotations for new (and possibly existing but touched) methods/classes. This would also prevent users from misusing these functions in the interim time while we add the annotations to the whole API (progressively, module-by-module, I guess).